apps/metadata-editor/src/app/records/records-list.component.ts

```diff
@@ -72,4 +77,27 @@ export class RecordsListComponent {
   showUsers() {
     this.router.navigate(['/users/my-org'])
   }
+
+  getSelectedRecords() {
+    return this.selectionService.selectedRecordsIdentifiers$
+  }
+
+  handleRecordsSelection(records: CatalogRecord[]) {
+    this.selectionService
+      .selectRecords(records)
+      .pipe(takeUntil(this.onDestroy$))
```

Comment:
Interesting, that looks like a valid approach to avoid having long-running observables stopping after a component is destroyed. Here though we shouldn't need it as the `selectRecords` and `deselectRecords` of the service are built on top of HTTP calls, which means they are short-lived observables and will complete after their first emission.